example/thirdparty/arrow/build.mill

```diff
@@ -0,0 +1,590 @@
+package build
```

Comment:
CC @lefou @lolgab in case you guys want to review this as well. We've never had a realistic kotlin multiplatform build before so this will be the flagship example.